hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AccountControllerTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.restjava.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.hedera.mirror.rest.model.NftAllowancesResponse;
+import com.hedera.mirror.restjava.RestJavaIntegrationTest;
+import com.hedera.mirror.restjava.mapper.NftAllowanceMapper;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@RequiredArgsConstructor
+class AccountControllerTest extends RestJavaIntegrationTest {
```

Comment:
This test class is testing at the wrong level. We want the controller to use Spring's `RestClient` to call the HTTP endpoint to ensure all levels of the http stack are verified. Don't extend `RestJavaIntegrationTest` and use `@WebMvcTest` instead.

```suggestion
@ExtendWith(SpringExtension.class)
@RequiredArgsConstructor
@WebMvcTest(controllers = AccountController.class)
class AccountControllerTest {
```

Reply:
Using RestClient with an actual @SpringBootTest which brings up a server.